code/go/0chain.net/smartcontract/minersc/fees.go

```diff
@@ -596,79 +469,18 @@ func (msc *MinerSmartContract) payShardersAndDelegates(fee, mint state.Balance,
 
 	// part for every sharder
 	for _, sh := range sharders {
-		var sresp string
-		if sh.numActiveDelegates() > 0 {
-			var delegateBr = state.Balance(float64(partm) * (1 - sh.ServiceCharge))
-			var delegateFees = state.Balance(float64(partf) * (1 - sh.ServiceCharge))
-			var sharderBR = state.Balance(float64(partm) * sh.ServiceCharge)
-			var sharderFees = state.Balance(float64(partf) * sh.ServiceCharge)
-
-			sresp, err = msc.payNode(sharderBR, sharderFees, sh, gn, balances)
-			if err != nil {
-				return "", err
-			}
-			resp += sresp
-
-			sresp, err = msc.payStakeHolders(delegateFees, sh, true, balances)
-			if err != nil {
-				return "", common.NewErrorf("pay_fees/pay_sharders",
-					"paying block sharder fees: %v", err)
-			}
-
-			resp += sresp
-
-			sresp, err = msc.mintStakeHolders(delegateBr, sh, gn, true, balances)
-			if err != nil {
-				return "", common.NewErrorf("pay_fees/mint_sharders",
-					"minting block sharder reward: %v", err)
-			}
-			resp += sresp
-		} else {
-			sresp, err = msc.payNode(partm, partf, sh, gn, balances)
-			if err != nil {
-				return "", err
-			}
-			resp += sresp
+		if err = sh.StakePool.DistributeRewards(
```

Comment:
We need to take the round value into account, suppose we have 3 sharders, and the fee is 5, the `partf` will be `1`, so the total fee distributed will be`3=len(sharders) * 1` , and  the remaining `2` will be burned. Same issue as the`mint.

Reply:
Removed type unnecessary type cast.